politeiad/backendv2/tstorebe/plugins/ticketvote/hooks.go

```diff
@@ -0,0 +1,370 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package ticketvote
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	backend "github.com/decred/politeia/politeiad/backendv2"
+	"github.com/decred/politeia/politeiad/backendv2/tstorebe/plugins"
+	"github.com/decred/politeia/politeiad/plugins/ticketvote"
+)
+
+// voteMetadataDecode decodes and returns the VoteMetadata from the
+// provided backend files. If a VoteMetadata is not found, nil is returned.
+func voteMetadataDecode(files []backend.File) (*ticketvote.VoteMetadata, error) {
+	var voteMD *ticketvote.VoteMetadata
+	for _, v := range files {
+		if v.Name != ticketvote.FileNameVoteMetadata {
+			continue
+		}
+		b, err := base64.StdEncoding.DecodeString(v.Payload)
+		if err != nil {
+			return nil, err
+		}
+		var m ticketvote.VoteMetadata
+		err = json.Unmarshal(b, &m)
+		if err != nil {
+			return nil, err
+		}
+		voteMD = &m
+		break
+	}
+	return voteMD, nil
+}
+
+func (p *ticketVotePlugin) linkByVerify(linkBy int64) error {
```

Comment:
doco please. Hard to read function

Reply:
Done
https://github.com/lukebp/politeia/blob/tlog/politeiad/backendv2/tstorebe/plugins/ticketvote/hooks.go#L124